Automattic/newspack-plugin
# Fixes Select Component initial value is always the first option

### All Submissions:






### Changes proposed in this Pull Request:

As detected by issue #83 , _"... when using the SelectControl component, the initial value for the select element upon page load uses the value of the first child option element, rather than having no - or an empty - value"_.

Here this is fixed by adding a first `disabled` option with the label "- Select -" and empty `value`. 

However, since the `SelectControl as BaseComponent` from`'@wordpress/components` doesn't support the `disabled` attribute for the `<option>` element (see [in codebase](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/select-control/index.js#L51)), now switched to using a manual `<select>` HTML element.

[This PR](https://github.com/WordPress/gutenberg/pull/15976) should eventually update the BaseComponent, but since a PR process could be relatively lengthy, and our publishers are experiencing the bug right now, currently switched the component to use a pure `<select>` element with a `<label>`.

Also as a minor change, in the Components Demo wizard, updated more meaningful labels and fixed the wrong use of onChange to handle event as argument, rather than value.


When now clicking on the Select component (specifically relevant for a Select with no previous selections), a first disabled option "- Select -" is present, enabling correct onClick of the first option:

![image](https://user-images.githubusercontent.com/29167323/61539526-0038e880-aa3c-11e9-89cb-9066f6163a5d.png)

Closes #83 .

### How to test the changes in this Pull Request:

1. Pull and build `nvm use 10.10.0 && npm ci && npm run clean && npm run build:webpack -- --watch`.
2. Visit the Components demo page at `/wp-admin/admin.php?page=newspack-components-demo`.
3. Verify that the three demoed Select Components are initially visually sound.
4. Check that the second Select component with label "Label for Select with no selection" now accepts selection of the first option with a single click, and that on clicking away the selection persists/sticks.
5. Test click the remaining two Select components.

### Other information:
